packages/tools/kadena-cli/src/account/utils/validateAccountDetails.ts

```diff
@@ -0,0 +1,61 @@
+import type { CommandResult } from '../../utils/command.util.js';
+import type {
+  IAccountDetailsResult,
+  IAddAccountManualConfig,
+} from '../types.js';
+import { compareConfigAndAccountDetails } from './compareConfigAndAccountDetails.js';
+import { createAccountName } from './createAccountName.js';
+import { getAccountDetailsFromChain } from './getAccountDetails.js';
+
+export interface IValidateAccountDetailsData {
+  isConfigAreSame: boolean;
+  config: IAddAccountManualConfig;
+  accountDetails: IAccountDetailsResult;
+}
+
+export async function validateAccountDetails(
+  config: IAddAccountManualConfig,
+): Promise<CommandResult<IValidateAccountDetailsData>> {
```

Comment:
`CommandResult` was only needed for the main logic method called from the cli command, not other helpers and utils. Making error messages should then also be moved up to the `addAccount` method